chemprop/data/datapoints.py

```diff
@@ -82,38 +60,54 @@ class MoleculeDatapoint(_DatapointMixin, _MoleculeDatapointMixin):
     ``d_vd`` is the number of additional descriptors that will be concatenated to atom-level
     descriptors *after* message passing"""
 
-    def __post_init__(self, mfs: list[MoleculeFeaturizer] | None):
+    def __post_init__(self):
         if self.mol is None:
             raise ValueError("Input molecule was `None`!")
 
         NAN_TOKEN = 0
-
         if self.V_f is not None:
             self.V_f[np.isnan(self.V_f)] = NAN_TOKEN
         if self.E_f is not None:
             self.E_f[np.isnan(self.E_f)] = NAN_TOKEN
         if self.V_d is not None:
             self.V_d[np.isnan(self.V_d)] = NAN_TOKEN
 
-        super().__post_init__(mfs)
+        super(MoleculeDatapoint, self).__post_init__()
```

Comment:
This is required if we have slots=True as this forces the dataclass to return a new class. https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses

I did a test where I made 10 M datapoints and saw the memory usage go from 1.6 G to 1.3 G, so I'd be okay not using slots=True.

Reply:
where do you see the part about `slots=True` in that link you posted?

Reply:
Sorry about that. I meant to send this link https://stackoverflow.com/questions/73268995/typeerror-when-calling-super-in-dataclassslots-true-subclass

Reply:
The memory difference seems negligible on that data scale. IMO we just remove `slots=True` to avoid some weird behavior coming back to bite us (like it does here--good work identifying this!)